.github/workflows/code-tidy.yml

```diff
@@ -0,0 +1,73 @@
+name: Tidy Checks
+
+on:
+  push:
+    branches:
+    - master
+  pull_request:
+    branches:
+    - master
+
+jobs:
+  RunClangTidy:
+    runs-on: ubuntu-24.04
+    steps:
+
+    - name: Checkout code
+      uses: actions/checkout@v4
+      with:
+        submodules: 'recursive'
+
+    - name: Install dependencies
+      run: |
+        sudo ./scripts/uni-get-dependencies.sh
+        sudo apt -qq -y install clang-tidy
+
+    - name: Run CMake
+      run: cmake -S . -B build -GNinja
+
+    - name: Build autogenerated files
+      run: ninja -C build CMakeFiles/OpenSCAD_autogen
+
+    - name: Create Cache Timestamp
+      id: cache_timestamp
+      uses: nanzm/get-time-action@v2.0
+      with:
+        format: 'YYYY-MM-DD-HH-mm-ss'
```

Comment:
Is the intent to get a unique build id? Timestamp is probably fine, but there might exist a build ID variable we could use. Fine for now though, just thought I'd ask..

Reply:
I deliberately choose a timestamp to have 'latest first' to be found by the cache. Subsequent runs will then find results that are most likely contain all relevant files.

The matching in finding the cache goes alphabetically if there is nothing found with an exact match, so that should work with that intention.

Having said that, there is room for improvement: if we are already in a pull request and have written the cache within that before, maybe we should re-use that very cache. Added a TODO.